src/modules/project/pages/projectView/hooks/useProjectGoals.tsx

```diff
@@ -48,6 +49,12 @@ export const useProjectGoals = (projectId: string | number | undefined) => {
     },
   })
 
+  const [updateProjectGoalOrdering] = useMutation(MUTATION_UPDATE_PROJECT_GOAL_ORDERING)
+
+  const handleUpdateProjectGoalOrdering = async (projectGoalIdsOrder: number[], projectId: string) => {
+    await updateProjectGoalOrdering({ variables: { input: { projectId, projectGoalIdsOrder } } })
+  }
+
```

Comment:
Once the ordering is completed, how do you refetch the goals or rather update the existing goals? The simple and elegant solution would be to have it in the `onCompletion` of the `useMutation(MUTATION_UPDATE_PROJECT_GOAL_ORDERING)`, for which we have the generated types too. why didn't you use the generated hooks for this?

Reply:
That's a good question, for this mutation I am not refetching the goals as they appear in the correct order in the state, they could be retched obviously if the user refreshes the page and display the correct order. I didn't think of an edge case that would require refetching the goals in this case juste like for `create` and `update` mutations. wdyt

Reply:
Hmm, that's a fair argument, the only case I can think of is, if the user sorts the values, moves to a different page inside the project context, i.e rewards list page or entries list page, then comes back to the goals, would the state persist for the dnd ?

Even if it did, it is volatile, as in the ordering that's shown is because of two things:
1. previous order of the goals.
2. dnd-context, of where each goal should be based on previous order. 

while after refetch or updating the state with new values ( without refetch), we're safe and we shouldn't consider this volatility if we're using this goal. i.e depends on one thing only:
1. Current order of the goal.

One another thing is, can the user sort the goals so the the default goal is changed? In that case would the change be reflected in the contribution summary section? (circular progress bar)

Reply:
This is a very good point Sajal, if the default goal is updated it indeed requires a project refetch to get the latest default goal id so the project balance display is properly updated.